### PR TITLE
refactor: Port security warnings to TypeScript

### PR DESCRIPTION
**Pull request by felixrieseberg**
Another chunk of TypeScript conversion, this time around for `security-warnings`.

Notes: no-notes

### CODE REVIEW

File: lib/renderer/security-warnings.ts [hunk 1]
```diff
@@ -1,6 +1,14 @@
-'use strict'
+import { webFrame } from 'electron'
+import { invokeSync } from '@electron/internal/renderer/ipc-renderer-internal-utils'
 
-let shouldLog = null
+declare global {
```
Comment (MarshallOfSound) [inline lib/renderer/security-warnings.ts]: Can this be put in our `ambient-*.d.ts` files so we don't have type mutations in individual files?

Comment (felixrieseberg) [reply lib/renderer/security-warnings.ts]: You can't mutate the global namespace from non-module files, so sadly not.

File: lib/renderer/security-warnings.ts [hunk 2]
```diff
@@ -73,7 +79,7 @@ const isUnsafeEvalEnabled = function () {
         return false
       }
       return true
-    }).toString()})()`, resolve)
+    }).toString()})()`, undefined, resolve)
```
Comment (MarshallOfSound) [inline lib/renderer/security-warnings.ts]: Is this a docs issue, or an issue in the code that typescript caught?

File: lib/renderer/security-warnings.ts [hunk 3]
```diff
@@ -264,17 +272,15 @@ const logSecurityWarnings = function (webPreferences, nodeIntegration) {
...
 const getWebPreferences = function () {
-  const ipcRendererUtils = require('@electron/internal/renderer/ipc-renderer-internal-utils')
-
   try {
-    return ipcRendererUtils.invokeSync('ELECTRON_BROWSER_GET_LAST_WEB_PREFERENCES')
+    return invokeSync('ELECTRON_BROWSER_GET_LAST_WEB_PREFERENCES')
   } catch (error) {
     console.warn(`getLastWebPreferences() failed: ${error}`)
-    return null
+    return undefined
```
Comment (MarshallOfSound) [inline lib/renderer/security-warnings.ts]: just `return` should work here right?

File: lib/renderer/security-warnings.ts (hunk 1 above)
Comment (MarshallOfSound) [reply lib/renderer/security-warnings.ts]: @felixrieseberg You can mutate the top level `Window` interface though

```
declare interface Window {
  hideLoader: () => void;
}
```

(copy pasted from another project)

File: lib/renderer/security-warnings.ts (hunk 2 above)
Comment (felixrieseberg) [reply lib/renderer/security-warnings.ts]: The latter 🙈

**Comment by MarshallOfSound** [comment]
@felixrieseberg Conflicting 😢

**Pull request merged**